core/src/main/java/org/jsmart/zerocode/core/runner/ZeroCodeMultiStepsScenarioRunnerImpl.java

```diff
@@ -102,8 +102,19 @@ public synchronized boolean runScenario(ScenarioSpec scenario, RunNotifier notif
 
             for (Step thisStep : scenario.getSteps()) {
 
-                final int stepLoopTimes = thisStep.getLoop() == null ? 1 : thisStep.getLoop();
+                boolean retryTillSuccess = false;
+                int delay = 0;
+                int stepLoopTimes = 1;
```

Comment:
Please do not mix loop and retries
User a different variable e.g. maxRetryAttempts to store number of retries